.gitignore

```diff
@@ -79,3 +79,6 @@ node_modules/
 
 # Generated JavaScript Bundles
 jsbundles
+
+# jenkins generated npm lock files at build time, however that should not be checked in ever
+package-lock.json
```

Comment:
the only reason I could see this happening is someone running `npm install`, maybe its ok to exclude it but we've never had a PR where someone has tried checking it in so I'm not convinced this is needed.
```suggestion
```

Reply:
https://github.com/jenkinsci/jenkins/pull/9757#discussion_r1768152697 IMHO it is cleaner to add it to the ignore just to make sure but I do not die on that hill

Reply:
https://github.com/jenkinsci/jenkins/pull/9757/files#r1768158913

Reply:
https://github.com/jenkinsci/jenkins/pull/9757/commits/5287bd55baa25261024fda560387097fab9f7e08